integration/integration_test.py

```diff
@@ -1048,10 +1114,33 @@ def file_to_stream(self, file_path, stream_path):
             print(' '.join(cmd))
         run_cmd(cmd)
 
+    def flight_request(self, json_path, arrow_path):
+        cmd = ['java', '-cp', self.ARROW_FLIGHT_JAR,
+               'org.apache.arrow.flight.example.integration.IntegrationTestClient',
+               '-j', json_path, '-a', arrow_path]
+        if self.debug:
+            print(' '.join(cmd))
+        run_cmd(cmd)
+
+    @contextlib.contextmanager
+    def flight_server(self):
+        cmd = ['java', '-cp', self.ARROW_FLIGHT_JAR,
+               'org.apache.arrow.flight.example.integration.IntegrationTestServer']
+        if self.debug:
+            print(' '.join(cmd))
+        server = subprocess.Popen(cmd)
+        try:
+            time.sleep(1)
```

Comment:
Yuck :-) This may be fragile, especially if you run the test on slow CI platforms.
Ideally you would wait on some event from the child process (for example it would print "Ready" after the server starts listening).

Reply:
Yes, good point. I should also have the port be controlled by the integration test script rather than just assuming some random port.